content/doc/book/glossary/index.adoc

```diff
@@ -5,15 +5,119 @@ layout: chapter
 :author:
 :email: jenkinsci-users@googlegroups.com
 :sectanchors:
-:toc: left
-:notitle:
+:toc:
+
+////
+XXX: Pages to mark as deprecated by this document:
+      * https://wiki.jenkins-ci.org/display/JENKINS/Terminology
+////
 
 [glossary]
 = Glossary
 
+////
+NOTE: The [glossary] delimiter in AsciiDoctor doesn't autogenerate anchors for
+each of the terms below. Which means that if we want to cross-reference terms
+directly from other documents we need to include an inline anchor.
+
+Additionally, because these inline anchors don't attach to section headings,
+cross referencing must include the appropriate display text, for example:
+
+  MyTerm:: [[myterm]] this is the definition of MyTerm
+
+Should be cross-referenced with:
+
+  <<myterm,MyTerm>>
+
+To ensure it is rendered appropriately.
+////
+
+== General Terms
 
 [glossary]
+Agent::  [[agent]]
+    An agent is typically a machine, or container, which connects to a Jenkins
+    master and executes tasks when directed by the master.
+Artifact:: [[artifact]]
+    An immutable file generated during a <<build,Build>> or <<pipeline,Pipeline>>
+    run. Artifacts are typically *archived* onto the Jenkins <<master,Master>> for
+    later retrieval by users.
+Build:: [[build]]
+    Result of a single execution of a <<project,Project>>
+Cloud:: [[cloud]]
+    A System Configuration which provides dynamic <<agent,Agent>>
+    provisioning and allocation, such as that provided by the
+    link:https://plugins.jenkins.io/azure-vm-agents[Azure VM Agents]
+    or
+    link:https://plugins.jenkins.io/ec2[Amazon EC2] plugins.
```

Comment:
I brought this over from [the terminology wiki page](https://wiki.jenkins-ci.org/display/JENKINS/Terminology). Since there's a button in "System Configuration" which says "Add Cloud" when one of those plugins which extends the extension point, I figured it would merit inclusion in the glossary.